### PR TITLE
Py101 - Lab02 - Lab material

### PR DESCRIPTION
**Pull request by sidereal-sky**

### CODE REVIEW

File: lab_02/task_1/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+* sa dubleze toate numerele care se divid cu 6, iar pe cele care nu, sa le tripleze;
+* sa transforme in litere mari toate vocalele dintr-o fraza
+* sa selecteze dintr-o lista doar cuvintele palindrom (se garanteaza ca toate cuvintele contin doar litere mici)
+
+## <ins>Exemplu</ins>
+
+```
+Task 1a
+input:
+[24, 3, 8, 18, 9, 6, 27]
```
Comment (RaduNichita) [inline lab_02/task_1/README.md:15]: Input and output should be without [ and ] (just the numbers).

I think it should be good to add the number of numbers as input

File: lab_02/task_1/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+> Pentru testare, puteți rula script-ul direct din IDE sau puteți rula în terminal comanda:
+
+```
+./checker.py
+```
+
+> In cazul in care nu puteti rula acest script din cauza permisiunilor, folositi mai intai urmatoarea comanda:
+
+```
+sudo chmod 700 checker.py
```
Comment (RaduNichita) [inline lab_02/task_1/README.md:58]:
```suggestion
sudo chmod 644 checker.py
```

File: lab_02/task_1/solved.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    '''
+    numere -> lista string-uri
+    return -> lista string-uri
+    '''
+
+    palindromes = filter(lambda cuv: cuv == cuv[::-1], cuvinte)
+
+    return list(palindromes)
+
+print(task1c(["demigod", "madam", "python", "php"]))
```
Comment (RaduNichita) [inline lab_02/task_1/solved.py:36]:
```suggestion
print(task1c(["demigod", "madam", "python", "php"]))

```

File: lab_02/task_1/task01a.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    Dublati elementele din vector si salvati-le
+    doar pe cele divizibile cu 3 folosind functionale
+    '''
+
+    ################### TO DO #########################
+    
+
+    ###################################################
+
+    # Nu modificati return-ul
```
Comment (RaduNichita) [inline lab_02/task_1/task01a.py:15]:
```suggestion
    # Nu modificati valoarea de retur a functiei
```

File: lab_02/task_1/task01c.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+def task1c(cuvinte):
+    '''
+    numere -> lista string-uri
+    return -> lista string-uri
+
```
Comment (RaduNichita) [inline lab_02/task_1/task01c.py:5]: palindromes = []

File: lab_02/task_3/task03.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    return -> lista doar cu numele studentilor
+    '''
+    nume = []
+
+    ################### TO DO #########################
+    
+
+    ###################################################
+    
+    return nume
```
Comment (RaduNichita) [inline lab_02/task_3/task03.py:13]:
```suggestion
    return nume
    
```

File: lab_02/task_2/solved.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+            result.append(elem)
+        elif type(elem) == str:
+            if elem not in "aeiou" and elem.islower():
+                result.append(elem)    
+        else:
+            continue
+    
+    return result
+
+print(task(50, 'A', 8.2, 'c', '_', 3, 'n', True, 'Z', 't', [1,2]))
```
Comment (RaduNichita) [inline lab_02/task_2/solved.py:19]:
```suggestion
print(task(50, 'A', 8.2, 'c', '_', 3, 'n', True, 'Z', 't', [1,2]))

```

File: lab_02/task_2/solved.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+def task(*args):
+    '''
+    numere -> elemente de mai multe tipuri
```
Comment (RaduNichita) [inline lab_02/task_2/solved.py:3]:
```suggestion
    numere -> elemente de tipuri diferite
```

File: lab_02/task_1/README.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+output:
+["madam", "php"]
+```
+
+> Indicatii de rezolvare:
+* folositi-va de functii lambda (cu if else sau fara) sau construiti-va functii aditionale
+* ajutati-va de un vector/string de vocale pt Task 1b
+
+> Explicatii return-uri in schelet:
+* dupa ce se aplica functionale, noul rezultat va fi de tipul "nume_functionala object"
```
Comment (RaduNichita) [inline lab_02/task_1/README.md:44]: you can give a link with for anyone curious to understand why map/filter/reduce doesn't return a list

File: lab_02/task_1/README.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+Gigel a inceput sa invete
+
+output:
+GIgEl A IncEpUt sA InvEtE
+```
+
+```
+Task 1c
+input:
+["demigod", "madam", "python", "php"]
```
Comment (RaduNichita) [inline lab_02/task_1/README.md:33]: the input and output should not contain [, ] or ".

File: lab_02/task_1/README.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+* folositi-va de functii lambda (cu if else sau fara) sau construiti-va functii aditionale
+* ajutati-va de un vector/string de vocale pt Task 1b
+
+> Explicatii return-uri in schelet:
+* dupa ce se aplica functionale, noul rezultat va fi de tipul "nume_functionala object"
+* folosim list pentru a converti rezultatul intors de functionala intr-o lista
+
+## <ins>Rulare | Testare</ins>
+
+> Pentru testare, puteți rula script-ul direct din IDE sau puteți rula în terminal comanda:
```
Comment (RaduNichita) [inline lab_02/task_1/README.md:49]: this is not a quote. I think you can leave it without >

File: lab_02/task_2/README.md [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+> Pentru testare, puteți rula script-ul direct din IDE sau puteți rula în terminal comanda:
+
+```
+./checker.py
+```
+
+> In cazul in care nu puteti rula acest script din cauza permisiunilor, folositi mai intai urmatoarea comanda:
+
+```
+sudo chmod 700 checker.py
```
Comment (RaduNichita) [inline lab_02/task_2/README.md:34]:
```suggestion
sudo chmod 644 checker.py
```

File: lab_02/task_2/task02.py [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+    return -> lista
+    '''
+    result = []
+
+    ################### TO DO #########################
+    
+
+    ###################################################
+    
+    return result
```
Comment (RaduNichita) [inline lab_02/task_2/task02.py:13]:
```suggestion
    return result
    
```

**Review by RaduNichita — CHANGES_REQUESTED** [review]

**Comment by Vlad1805** [comment]
Adauga un director skel in care sa fie task1, task2 si task3.
Muta fisierele de solved.py intr-un director solved.

**Pull request merged**